### ISSUE TITLE
Add TensorBoard Visualization

### ISSUE THREAD

**Issue by RobotSail** [instructlab/training#360]
Today we output JSON logs containing statistics about training runs, but it would be nice for local experiment tracking to be able to output various values like the training loss, validation loss (if/when we add one), gradnorm, etc.

Wandb helps solve this issue but TensorBoard is already built into PyTorch and something that can be easily viewed within VSCode.

**Comment by booxter** [comment]
This is relevant: https://github.com/instructlab/training/pull/500

Apart from collecting logs, is there anything else to be done in the library for visualization?

**Comment by RobotSail** [comment]
@booxter Nope, once you have pip installed tensorboard, you can just spin it up locally and point it to where you're saving the logs, and it works its magic from there

**Comment by RobotSail** [comment]
@booxter In the future we should consider adding more values that we log out, and when we do this I think we should make it simpler to group different groups together. For instance, training loss, validation loss (if present), etc. should all be grouped with each other for example. Similar for other values like regularization penalties, weight norms, etc.

**Comment by booxter** [comment]
@RobotSail these are interesting suggestions. Should they be tracked in this issue or should they go in a separate enhancement? (I think the latter since it would require not just visualization work but deciding on which metrics / events should be exposed etc.)

**Comment by RobotSail** [comment]
@booxter Separate issue
